padertorch/configurable.py

```diff
@@ -995,17 +1066,44 @@ def config_to_instance(config, strict=False):
     Tried to instantiate/call <class 'torch.nn.modules.activation.ReLU'> with
     `torch.nn.modules.activation.ReLU(**{'inplace_typo': False})`.
     Signature: (inplace=False)
+    >>> config = {
+    ...     'partial': 'torch.nn.modules.activation.ReLU',
+    ...     'inplace': False}
+    >>> config_to_instance(config)
+    functools.partial(<class 'torch.nn.modules.activation.ReLU'>, inplace=False)
+    >>> config_to_instance(config, strict=True)
+    functools.partial(<class 'torch.nn.modules.activation.ReLU'>, inplace=False)
+    >>> config = {
+    ...     'partial': 'torch.nn.modules.activation.ReLU'}
+    >>> config_to_instance(config)
+    <class 'torch.nn.modules.activation.ReLU'>
 
     """
     if isinstance(config, dict):
-        if 'factory' in config:
-            factory, kwargs = _split_factory_kwargs(config)
-            factory = import_class(factory)
+        special_key = _get_special_key(config)
+        if special_key:
+            factory, kwargs = _split_factory_kwargs(config, key=special_key)
+            if isinstance(factory, str):
+                factory = import_class(factory)
+            elif callable(factory):
+                pass
+            else:
+                raise TypeError(f'The special key {special_key} expects a'
+                                f' string or a callable but got',
+                                type(factory), factory)
```

Comment:
You could here also do (after changing the assertion to a `TypeError` in `import_class`)
```python
try:
    factory = import_class(factory)
except TypeError as e:
    raise TypeError(...) from e
```
It is a little bit confusing that this check is made twice. But, the `try`/`except` variant might create confusing errors if a `TypeError` is raised inside `import_class` that is not caused by the type of factory.

Reply:
I am partial to both solutions. At the moment I do not see a possibility for a TypeError inside of import_class that is not caused by the factory type. So I will change this to `try / except`